Source/NSViewController.m

```diff
@@ -179,6 +190,48 @@ - (void) encodeWithCoder: (NSCoder *)aCoder
       [aCoder encodeObject: [self view]];
     }
 }
+
+
+// NSSeguePerforming methods...
+- (void)performSegueWithIdentifier: (NSStoryboardSegueIdentifier)identifier 
+                            sender: (id)sender
+{
+  BOOL should = [self shouldPerformSegueWithIdentifier: identifier
+                                                sender: sender];
+
+  if (should)
+    {
+      NSStoryboardSegue *segue = [_segueMap objectForKey: identifier];
+      id destCon = nil;
+      if ([[segue destinationController] isKindOfClass: [NSViewController class]] ||
+          [[segue destinationController] isKindOfClass: [NSWindowController class]])
+        {
+          destCon = [segue destinationController];
+        }
+      else
+        {
+          NSString *destId = [segue destinationController];
+          destCon = [_storyboard instantiateControllerWithIdentifier: destId]; 
```

Comment:
Where is _storyboard ever set?

Reply:
`      
      while ((o = [en nextObject]) != nil)
        {
          if ([o isKindOfClass: [NSWindowController class]])
            {
              wc = (NSWindowController *)o;
              [wc _setSegueMap: segueMap];
              [wc _setTopLevelObjects: topLevelObjects];
              [wc _setStoryboard: self];
              [wc _setOwner: NSApp];
              result = o;`

Reply:
NSStoryboard.m in the _processStoryboard: method.

Reply:
My feeling here is that most of this code actually belongs into the segue. But to be sure I should be able to better understand all your transformation code to see whether stuff could be easily moved around. Why doesn't the segue know about the owning story board and and computes it destination controller all by itself?

Reply:
The only time the code can get the destination controller is when the segue is initiated.  Until then it's necessary to simply store the id.

Reply:
I don't think this is needed at this point. ;)